### ISSUE TITLE
[SDL3] Pen  - IOS pen detecting as mouse

### ISSUE THREAD

**Issue by tegaologe** [libsdl-org/SDL#9911]
does anyone have solution for IPadOS pencil registering as SDL_EVENT_MOUSE_MOTION, i can't get it to detect SDL_EVENT_PEN_MOTION

Stuck trying to figure it out

**Comment by icculus** [comment]
Pen support hasn't been implemented for iOS yet.

**Comment by coinbump** [comment]
This is an issue on MacOS as well. Not sure if they're related.

**Comment by coinbump** [comment]
On Mac,i all we need is the pressure, we can get that like so:

```
void Cocoa_HandleMouseEvent(SDL_VideoDevice *_this, NSEvent *event)
{
    float pressure = event.pressure;
```

So if we could just add a way to send mouse button pressure, independently of pen events that would be useful.

**Comment by icculus** [comment]
This has since been implemented for macOS; iOS is still pending.

**Comment by frenzibyte** [comment]
@icculus Hi, would it be okay if I take a jab at this? It would be nice to have native Apple Pencil support for our game, mostly for its support of hovering over components while being recognized as a pen instead of a mouse. I've tried going through this to see what needs to be done before blindly jumping and asking, and there does seem to be a path forward to have it work without breaking `UIPointerInteraction`-based mouse support, by using `UIHoverGestureRecognizer` filtered to apple pencil events specifically. If you're interested, I've pushed a WIP commit for viewing the changes here: https://github.com/libsdl-org/SDL/commit/27875bc7dd4a5d99600d2145a1485173a9a83d59

The only part left from my changes above is translating Apple Pencil's altitude and azimuth coordinates to tilt-x/y for SDL pen axis events. I guess I'll have to figure out the vector math behind it 😅

**Comment by icculus** [comment]
This looks really good so far, but there are a few minor tweaks that are needed; I'm going to pull this commit in tonight (so you get credit for the heavy lifting in the commit history) and work on top of it, if you don't mind.

**Comment by frenzibyte** [comment]
Absolutely, I'm glad I got somewhere.

**Comment by icculus** [comment]
This should be resolved now (pen and mouse and touch events are all separate on iOS); let me know if it's not, and we'll reopen!
